rcljava/src/main/java/org/ros2/rcljava/publisher/PublisherImpl.java

```diff
@@ -101,6 +111,42 @@ public final long getHandle() {
     return this.nodeReference;
   }
 
+  /**
+   * {@inheritDoc}
+   */
+  public final
+  <T extends PublisherEventStatus> EventHandler<T, Publisher>
+  registerEventHandler(Supplier<T> factory, Consumer<T> callback) {
+    T status = factory.get();
+    long eventHandle = nativeCreateEvent(this.handle, status.getPublisherEventType());
+    EventHandler<T, Publisher> eventHandler = new EventHandlerImpl(
+      new WeakReference<Publisher>(this), eventHandle, factory, callback);
+    this.eventHandlers.add(eventHandler);
+    return eventHandler;
+  }
+
+  /**
+   * {@inheritDoc}
+   */
+  public final
+  <T extends PublisherEventStatus> void disposeEventHandler(
```

Comment:
bike shed: "remove" sounds more natural

```suggestion
  <T extends PublisherEventStatus> void removeEventHandler(
```

Reply:
Fixed in https://github.com/osrf/ros2_java/pull/9/commits/372edb1544f7ea8a838f6717399f9b9407cc476c.